python_modules/dagster/dagster/core/execution.py

```diff
@@ -583,33 +589,89 @@ def construct_pipeline_execution_context(
     )
 
 
-def _create_loggers(run_config, execution_context):
+def _create_loggers(environment_config, run_config, pipeline_def):
+    check.inst_param(environment_config, 'environment_config', EnvironmentConfig)
     check.inst_param(run_config, 'run_config', RunConfig)
-    check.inst_param(execution_context, 'execution_context', ExecutionContext)
+    check.inst_param(pipeline_def, 'pipeline_def', PipelineDefinition)
+
+    loggers = []
+    for logger_key, logger_def in pipeline_def.loggers.items() or default_loggers().items():
+        if logger_key in environment_config.loggers:
+            loggers.append(
+                logger_def.logger_fn(
+                    InitLoggerContext(
+                        environment_config.context,
+                        environment_config.loggers.get(logger_key),
+                        pipeline_def,
+                        logger_def,
+                        run_config.run_id,
+                    )
+                )
+            )
+
+    if run_config.loggers:
+        for logger in run_config.loggers:
+            loggers.append(logger)
+
+    if not loggers:
+        for (logger_def, logger_config) in default_system_loggers():
```

Comment:
also what is different between default loggers and default system loggers?

Reply:
see above